### PR TITLE
entity detection logic and meshkit errors w/signoff

### PR DESCRIPTION
**Pull request by Jougan-0**
**Description**

This PR fixes #

**Notes for Reviewers**

**[Signed commits](../CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [ ] Yes, I signed my commits.

### CODE REVIEW

**Comment by MUzairS15** [comment]
My reviews didn't get submitted. 🥲

File: utils/utils.go [hunk 1]
```diff
@@ -430,3 +431,22 @@ func IsSchemaEmpty(schema string) (valid bool) {
...
+	if schemaVersion, ok := tempMap["schemaVersion"].(string); ok {
+		lastIndex := strings.LastIndex(schemaVersion, "/")
+		if lastIndex != -1 {
+			schemaVersion = schemaVersion[:lastIndex]
+		}
+		switch schemaVersion {
+		case "relationships.meshery.io":
+			return entity.RelationshipDefinition, nil
+		case "components.meshery.io":
+			return entity.ComponentDefinition, nil
```
[the remark this answers is not in the document]
Comment (Jougan-0) [reply utils/utils.go:448]: Adding them in next .

File: utils/utils.go [hunk 2]
```diff
@@ -430,3 +431,22 @@ func IsSchemaEmpty(schema string) (valid bool) {
...
+		}
+	}
+	return "", nil
```
[the remark this answers is not in the document]
Comment (Jougan-0) [reply utils/utils.go:451]: if we return error it was breaking the handler I'll update the handler and then push the PR.

**Comment by Jougan-0** [comment]
> My reviews didn't get submitted. 🥲

It slipped my bad will update them.

**Pull request merged**